Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,188 @@
+            inputGraphicsOverlay.renderer = renderer
+            return inputGraphicsOverlay
+        }()
+        
+        /// Removes previously tapped location from overlay and draws new dot on tap location.
+        /// - Parameter point: Location to add a dot graphic on the map.
+        func addInputGraphic(at point: Point) {
+            inputGraphicsOverlay.removeAllGraphics()
+            let graphic = Graphic(geometry: point)
+            inputGraphicsOverlay.addGraphic(graphic)
+        }
+        
+        /// Controls the initialization for the geoprocessing of the viewshed and calls the geoprocessing logic.
+        /// - Parameter point: Location that the user tapped on the map.
+        func calculateViewshed(at point: Point) async throws {
+            // Clears previously viewshed drawing.
+            resultGraphicsOverlay.removeAllGraphics()
+            // If there is a geoprocessing job in progress it is cancelled.
+            await geoprocessingJob?.cancel()
+            guard let spatialReference = point.spatialReference else { return }
+            // Creates a feature collection table based on the spatial reference for the tapped location
+            // on the map.
+            let featureCollectionTable = FeatureCollectionTable(
+                fields: [Field](),
+                geometryType: Point.self,
+                spatialReference: spatialReference
+            )
+            // Creates a feature for the point tapped.
+            let feature = featureCollectionTable.makeFeature(geometry: point)
+            // Asynchronously adds that feature to the table.
+            try await featureCollectionTable.add(feature)
+            try await performGeoprocessing(featureCollectionTable)
+        }
+        
+        /// Contains the logic for the geoprocessing and passes the result on to another function to display.
+        /// - Parameter featureCollectionTable: Holds the tapped location feature
+        private func performGeoprocessing(_ featureCollectionTable: FeatureCollectionTable) async throws {
+            let params = GeoprocessingParameters(executionType: .synchronousExecute)
+            // Sets the parameters spatial reference to the point tapped on the map.
+            params.processSpatialReference = featureCollectionTable.spatialReference
+            params.outputSpatialReference = featureCollectionTable.spatialReference
+            // Create a feature with the feature collection table which has the reference to the tapped location.
+            let geoprocessingFeature = GeoprocessingFeatures(features: featureCollectionTable)
+            // Sets the observation point to the tapped location.
+            params.setInputValue(geoprocessingFeature, forKey: "Input_Observation_Point")
+            // Creates geoprocessing job and kicks off process of getting viewshed for the point.
+            geoprocessingJob = geoprocessingTask.makeJob(parameters: params)
```

Comment:
I don't think the [comment](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/427#discussion_r1639053258) is fully addressed.

- job reference can be set to `nil` after it finishes. You can defer that
- either hold a local job reference, or force unwrap the `geoprocessingJob` so it doesn't get conditionally chained 3 times - `geoprocessingJob?.start()`, `geoprocessingJob?.output` and `output?.outputs["Viewshed_Result"]`

Reply:
Should be resolved now per the previous comments suggestions.